rfcs/rfcOVID-0019.md

```diff
@@ -0,0 +1,340 @@
+    our %EXPORT_TAGS = qw(
+        'all'     => \@EXPORT_OK,
+        'strings' => qw(bay baz),
+    );
+
+    sub bar  {...}
+    sub baz  {...}
+    sub bay  {...}
+    sub quux {...}
+    sub whee {...}
+}
+```
+
+The requested functions are imported lexically via the
+`builtin::export_lexically` function, rather than appearing in the package
+symbol table. This means that code within that lexical scope can see and call
+them as normal, but they are not visible from outside the scope to which they
+are exported, such as from other callers or as object or class methods. This
+removes the need for `namespace::clean` and related modules.
+
+Upper-case `:export` attribute arguments are reserved for Perl to avoid
+clashing with user-defined tags. This is because in this author's experience,
+export tags are _usually_ lower-case.
+
+### Interface
+
+The various incantations of `:export` should populate the respective
+`@EXPORT`, `@EXPORT_OK`, and `%EXPORT_TAGS` package variables to maintain a
+consistent API with the `Exporter` module.
+
+The grammar for the tag is:
+
+    <export>  ::= ':export' [ <options> ]
+    <options> ::=  '(' <tags> ')'
+    <tags>    ::=  <tag> { ',' <tag> }
+    <tag>     ::=  'DEFAULT' | \w+
+
+
+### Exporting
+
+* `sub foo :export                  {}` Export only on demand
+* `sub foo :export(DEFAULT)         {}` Always export
+* `sub foo :export(strings)         {}` Export with `:strings` tag or directly
+* `sub foo :export(DEFAULT,strings) {}` Export with `:strings` tag or automatically
+* `sub foo :export(as,df)           {}` Export with either the `:as` or `:df` tags, or directly
+
+Any function that has a tag or tags (such as `strings` in the above example),
+can also be imported directly by name.
+
+If a function has a tag, such as in the `:export(strings)` example, it's
+automatically a function which can be exported on demand. If you wish that
+function to be exported automatically if the tag is not supplied, you must
+also use the special `DEFAULT` tag: `:export(DEFAULT,strings)`.
+
+### Importing
+
+* `use Mod;` Import only functions declared with `:export(DEFAULT)`
+* `use Mod qw(foo bar baz)` Import functions iff declared with `:export`
+* `use Mod ':strings';` Import only functions tagged with `strings`
+* `use Mod ':all';` All function with any form of `:export` tag are imported
+* `use Mod qw(:all !foo);` Import all functions except `foo`.
+
+Note that the venerable `Exporter` module also allows importing based on
+patterns:
+
+```perl
+use Socket qw(!/^[AP]F_/ !SOMAXCONN !SOL_SOCKET);
+use POSIX  qw(:errno_h :termios_h !TCSADRAIN !/^EXIT/);
+```
+
+This RFC does not include support for this.
+
+### Implementation
+
+The `builtin::export_lexically` function will be used to provide the actual
+functionality. `:export` tags, if used, their options, will be used to decide
+which functions used to decide which functions are lexically exported.
+
+This module will not allow unexporting lexical functions.
```

Comment:
Should this say "This module will not allow exporting lexical functions."?

If so, why not?  The main issue is it doesn't work with `@EXPORT_OK`, since that only includes names, but I can see it being useful to export a sub that you don't want to be a method.

Also, should that be "RFC" instead of "module"?

Reply:
@tonycoz Thank you. I've fixed the wording and pushed.